### PR TITLE
Inverted Index Application Update

### PR DESCRIPTION
**Pull request by abdulsemiu-atanda**
### What this PR do?
Implement feedback from PR #1 

### How to manually test this PR?
Check Files changed

### CODE REVIEW

File: src/inverted-index.js [hunk 1]
```diff
@@ -1,85 +1,95 @@
...
-                        }
-                        
-                        wordOccurrence[key].push(i);
-                    }
-                }
+    if (file.length === 0) {
+      return false;
+    }
+    else {
+
```
Comment (q-ode) [inline src/inverted-index.js]: Empty line here...

File: src/inverted-index.js [hunk 2]
```diff
@@ -1,85 +1,95 @@
...
+  /**
+   * Get Index method that indexes JSON file
+   * @param data (key,value pair)
+   * @return fileIndex
+   */
+  getIndex(data) {
+    let fileIndex = {};
+
+    for (let i = 0; i < data.docs.length; i++) {
+
```
Comment (q-ode) [inline src/inverted-index.js]: Empty line...

File: src/inverted-index.js [hunk 3]
```diff
@@ -1,85 +1,95 @@
...
-                        
-                        wordOccurrence[key].push(i);
-                    }
-                }
+    if (file.length === 0) {
+      return false;
+    }
+    else {
+
+      for (let i = 0; i < file.length; i++) {
```
Comment (q-ode) [inline src/inverted-index.js]: Nitpick: What is i?

File: src/inverted-index.js [hunk 4]
```diff
@@ -1,85 +1,95 @@
...
-    searchIndex(terms,reff) {
-        let result = {};
-        if (typeof terms !== 'string' || terms === '' || (terms.trim()) === '') {
-            return false;
+    else {
+      const token = tokenize(terms);
+      const words = token.split(' ');
+
+      for (let i = 0; i < words.length; i++) {
+
```
Comment (q-ode) [inline src/inverted-index.js]: Empty space...

File: src/inverted-index.js [hunk 5]
```diff
@@ -1,85 +1,95 @@
...
+        if (occurrence[words[i]] === undefined) {
+
```
Comment (q-ode) [inline src/inverted-index.js]: Space...

File: src/pageCtrl.js [hunk 6]
```diff
@@ -1,54 +1,78 @@
...
+
+  };
+
+  $scope.index = (fileIndex) => {
+    $scope.docs = $scope.documents[fileIndex].docs;
+    if ($scope.docs.length < 1) {
+      $scope.docArray = [];
+      $scope.message = 'Your File is empty';
+    }
+
```
Comment (q-ode) [inline src/pageCtrl.js]: Space...

File: src/pageCtrl.js [hunk 7]
```diff
@@ -1,54 +1,78 @@
...
-        return false;
-    }
-    return true;
-};
\ No newline at end of file
+const isJson = (str) => {
+  try {
+    JSON.parse(str);
+  } catch (e) {
+    return false;
```
Comment (q-ode) [inline src/pageCtrl.js]: Not properly indented.

File: src/inverted-index.js [hunk 8]
```diff
@@ -1,85 +1,95 @@
...
-    /**
-     * Search Index Method that searches uploaded file with the aid of the created index
-     * @param terms(words to be searched), reff
-     * @return result
-     */
-    searchIndex(terms,reff) {
-        let result = {};
-        if (typeof terms !== 'string' || terms === '' || (terms.trim()) === '') {
-            return false;
+    else {
```
Comment (q-ode) [inline src/inverted-index.js]: AirBnB Style Guide: `put else on the same line as your if block's closing`

**Review by q-ode — CHANGES_REQUESTED** [review]
Kindly go through the AirBnB Style Guide once again and see that you conform to these recommendations.
https://github.com/airbnb/javascript

File: src/pageCtrl.js (hunk 6 above)
Comment (abdulsemiu-atanda) [reply src/pageCtrl.js]: @q-ode I was following the space after every block rule of Airbnb

File: src/pageCtrl.js [hunk 9]
```diff
@@ -1,54 +1,79 @@
 'use strict';
-let app = angular.module('app',[]);
+/**
+ * Checks if contents of the file is in JSON format
+ */
+const isJson = (term) => {
```
Comment (q-ode) [inline src/pageCtrl.js]: Is this supposed to be a utility function?
If so, it shouldn't be here. Also, why are you declaring functions as `const`?

File: src/inverted-index.js [hunk 10]
```diff
@@ -1,85 +1,89 @@
...
+ * @param {String} tokens
+ * @return {Array} newToken
  */
-    getIndex(data) {
-        let fileIndex = {};
-        for(let i=0;i<data.docs.length;i++) {
-            fileIndex[data.name] = [i];
-        }
-        return fileIndex;
+const tokenize = (tokens) => {
```
Comment (q-ode) [inline src/inverted-index.js]: I believe you're trying to declare a class function here.

Firstly, it shouldn't be outside the class.

Secondly, why are you declaring functions with the keyword `const`?

File: src/inverted-index.js [hunk 11]
```diff
@@ -1,85 +1,89 @@
...
+  const newToken = tokens.replace(/[^a-z0-9]+/gi, ' ')
+    .trim().toLowerCase();
+  // send out the new token
+  return newToken;
```
Comment (q-ode) [inline src/inverted-index.js]: The variable `newToken` is redundant here. You're declaring it just to return the value.

What do you think about returning said value directly?

File: src/inverted-index.js [hunk 12]
```diff
@@ -1,85 +1,89 @@
...
+    }
+    const token = tokenize(terms);
+    const words = token.split(' ');
+    for (let count = 0; count < words.length; count += 1) {
+      if (occurrence[words[count]] === undefined) {
+        result[words[count]] = 'Not Found';
+      } else {
+        const key = words[count];
+        const value = occurrence[words[count]];
+        result[key] = value;
```
Comment (q-ode) [inline src/inverted-index.js]: The use of variable `value` is redundant here as you only initialized it in the line above just to pass the value to this line.

**Review by q-ode — CHANGES_REQUESTED** [review]
Hey man,
Kindly attend to the changes and reach out to me if you need further clarification. 

PS: Why is your setup of HoundCI throwing so many errors?

**Pull request merged**